# Comparación entre dos valores esperados poblacionales

In [ ]:
!pip install pingouin

In [ ]:
import pandas as pd
from statsmodels.stats.weightstats import ztest as ztest
import pingouin as pg

In [ ]:
df = pd.read_excel("/content/historial_clientes.xlsx")

In [ ]:
df.head()

Cliente A  Cliente B
0   1.247098   2.095465
1   0.936653   1.484473
2  -0.186285   1.744021
3   1.494336   1.683304
4   2.557917   2.306799

**Prueba Z**
 Hipótesis nula
 $$ H_0: \mu_1 = \mu_2$$

 Estadistico de prueba
 $$ Z = \frac{\overline{x} - \overline{y} - (\mu_1 - \mu_2)}{\sqrt{\frac{\sigma_1^2}{m}+\frac{\sigma_2^2}{n}}}$$

 El estadistico de prueba tiene una distribución estándar.

**Realizar el ztest**

In [ ]:
ztest(df["Cliente A"], df["Cliente B"])

(-2.6683380460572605, 0.007622751887296756)

# Prueba T

## Distribución T-Student

In [ ]:
url = ("https://raw.githubusercontent.com/JoaquinAmatRodrigo/Estadistica-machine-learning-python/master/data/births.csv")
datos = pd.read_csv(url, sep = ",")
datos.head()

f_age  m_age  weeks  premature  visits  gained  weight sex_baby      smoke
0   31.0     30     39  full term    13.0     1.0    6.88     male     smoker
1   34.0     36     39  full term     5.0    35.0    7.69     male  nonsmoker
2   36.0     35     40  full term    12.0    29.0    8.88     male  nonsmoker
3   41.0     40     40  full term    13.0    30.0    9.00   female  nonsmoker
4   42.0     37     40  full term     NaN    10.0    7.94     male  nonsmoker

 **Prueba T-student**
 Hipótesis nula
 $$ H_0: \mu_1 = \mu_2$$

 Estadistico de prueba

 $$ T = \frac{\overline{x} - \overline{y} - (\mu_1 - \mu_2)}{\sqrt{\frac{S_1^2}{m}+\frac{S_2^2}{n}}}$$

El et

Filtrando entre madres fumadoras y no  fumadoras.

In [ ]:
peso_smokers = datos.loc[datos.smoke == "smoker", "weight"]
peso_nonsmokers = datos.loc[datos.smoke == "nonsmoker", "weight"]

In [ ]:
peso_smokers

In [ ]:
peso_nonsmokers

Aplicando t-test

In [ ]:
pg.ttest(x = peso_smokers, y = peso_nonsmokers)

T        dof alternative     p-val          CI95%   cohen-d  \
T-test -1.496711  89.277413   two-sided  0.137995  [-0.93, 0.13]  0.268758   

         BF10     power  
T-test  0.512  0.338075

Como p-valor = 0.137995 > 0.05
T = -1.4 está en la R. de aceptación.
Es decir $\mu_1 = \mu_2$